src/GafferScene/Instancer.cpp

```diff
@@ -2796,34 +2899,38 @@ void Instancer::InstancerCapsule::render( IECoreScenePreview::Renderer *renderer
 	// Output the instances
 	// ============================================================================
 
-	const std::vector<int> &pointIndicesForPrototype = engines[0]->pointIndicesForPrototype( root().back() );
-
 	// We've found problems with performance when running too many iterations in parallel, which appear
 	// to be related with hitting AiNode too hard in parallel ( perhaps related to threads spread between
 	// separate processors ). To partially solve this, we set the grain size so that we shouldn't use more
 	// than 32 threads, which appears to help some in testing.
-	size_t grainSize = std::max( (size_t)1, pointIndicesForPrototype.size() / 32 );
-	task_group_context taskGroupContext( task_group_context::isolated );
+	size_t grainSize = std::max( (size_t)1, engines[0]->numPoints() / 32 );
 
-	const ThreadState &threadState = ThreadState::current();
-	tbb::parallel_for( tbb::blocked_range<size_t>( 0, pointIndicesForPrototype.size(), grainSize ),
+	tbb::parallel_for( tbb::blocked_range<size_t>( 0, engines[0]->numPoints(), grainSize ),
 		[&]( const tbb::blocked_range<size_t> &r )
 		{
 			Context::EditableScope prototypeScope( threadState );
 
 			vector<M44f> pointTransforms( sampleTimes.size() );
-			std::string name;
 			IECoreScenePreview::Renderer::AttributesInterfacePtr attribsStorage;
 
+			// Storage for names, indexed by prototype id ( each instance of the same prototype
+			// will reuse this storage )
+			std::vector< std::string > names( engines[0]->numValidPrototypes() );
+			std::vector< int > namePrefixLengths( engines[0]->numValidPrototypes() );
 
-			for( size_t idx = r.begin(); idx != r.end(); ++idx )
+			for( size_t i = r.begin(); i != r.end(); ++i )
```

Comment:
Maybe `i` could be called `pointIndex`, replacing the original variable? Then less code would change, and I think it'd be a bit easier to follow anyway...